### PR TITLE
Make restriction lint's use `span_lint_and_then` (i -> p)

### PR DESCRIPTION
**Pull request by xFrednet**
This migrates a few restriction lints to use `span_lint_and_then`. This change is motivated by https://github.com/rust-lang/rust-clippy/issues/7797.

I've also cleaned up some lint message. Mostly minor stuff. For example: suggestions with a longer message than `"try"` now use `SuggestionStyle::ShowAlways`

---

cc: https://github.com/rust-lang/rust-clippy/issues/7797

brother PR of: https://github.com/rust-lang/rust-clippy/pull/13136

changelog: none

### CODE REVIEW

**Comment by y21** [comment]
Is the plan to completely move away from these other `span_lint_and_*` functions? I feel like we should just remove that internal lint then 🤔. Seems odd that we require using functions that shouldn't be used or that we have to add #[expect]s everywhere.

I really liked `span_lint_and_{sugg,help}` more for when the messages are just plain string literals and that seems like it shouldn't make a difference there, but oh well

**Comment by xFrednet** [comment]
> Is the plan to completely move away from these other `span_lint_and_*` functions?

That's a good question. The issue suggests the entire removal of these functions to have one unified way. I also think that rustc's diagnostic documentation is pretty good, which would make it easier for new contributors.

For restriction lints, I think it makes highly sense to remove them. Lints like [`implicit_return`](https://rust-lang.github.io/rust-clippy/master/index.html#/implicit_return) had previously created a suggestion for every implicit return. I slightly changed the message, and that's where lintcheck got the 60000+ changes from.

Now, I've kept the internal lint around, as I'm unsure, how motivated I am to migrate all of Clippy. These three PRs represent ~116 of our 700+ lints. Even if we create a migration lint it would take dedication. And rn I'm not sure if I'm committed enough for that. If we don't migrate all of Clippy, it would be better to keep the lint for consistency.

File: clippy_lints/src/methods/map_err_ignore.rs [hunk 1]
```diff
@@ -22,13 +22,18 @@ pub(super) fn check(cx: &LateContext<'_>, e: &Expr<'_>, arg: &Expr<'_>) {
...
+        span_lint_and_then(
             cx,
             MAP_ERR_IGNORE,
             fn_decl_span,
             "`map_err(|_|...` wildcard pattern discards the original error",
-            None,
-            "consider storing the original error as a source in the new error, or silence this warning using an ignored identifier (`.map_err(|_foo| ...`)",
+            |diag| {
+                diag.help(
+
```
Comment (y21) [inline clippy_lints/src/methods/map_err_ignore.rs:33]: This extra newline looks like it can be removed

File: clippy_lints/src/float_literal.rs [hunk 2]
```diff
@@ -109,29 +109,41 @@ impl<'tcx> LateLintPass<'tcx> for FloatLiteral {
                     // If the type suffix is missing the suggestion would be
```
Comment (y21) [inline clippy_lints/src/float_literal.rs:109]: Looks like this comment here was meant to be part of the `if` that was moved down

**Review by y21 — APPROVED** [review]
Looks good, just two small "formatting" nits.

**Comment by xFrednet** [comment]
Very good catches! Thank you for taking the time =^.^=

The last commit, should have resolved everything.

**Comment by y21** [comment]
Looks all good to me now! Want to squash the last commit or keep it? You can `r=me` either way

**Comment by xFrednet** [comment]
And squashed :D

Thank you for the quick feedback =^.^=

---

Roses are red,
Violets are blue,
This has been approved,
by the one and only Timo

**Pull request merged**